### PR TITLE
feature/distage-docker-network-lifecycle

### LINKED ISSUE #946: distage-framework-docker: randomize network names
Individual container names are randomized, however there's no option to randomize network names – they should also be randomized to avoid conflicts.

/cc @Caparow

### PR DESCRIPTION
**Pull request by Caparow**
Fix #946

### CODE REVIEW

File: distage/distage-framework-docker/src/main/scala/izumi/distage/docker/DockerContainer.scala [hunk 1]
```diff
@@ -87,14 +88,30 @@ object DockerContainer {
...
   private[this] final case class PortDecl(port: DockerPort, localFree: Int, binding: PortBinding, labels: Map[String, String])
 
-  final class Resource[F[_]: DIEffect: DIEffectAsync, T](
+  final case class ContainerResource[F[_], T](
     config: Docker.ContainerConfig[T],
     clientw: DockerClientWrapper[F],
     logger: IzLogger,
+  )(
+    implicit
+    val ef: DIEffect[F],
```
Comment (neko-kai) [inline distage/distage-framework-docker/src/main/scala/izumi/distage/docker/DockerContainer.scala]: you may name `F` & `P` and in that case you can replace `DIEffect[F].method` below by `F.method`, `Async.sleep`->`P.sleep`

File: distage/distage-framework-docker/src/main/scala/izumi/distage/docker/ContainerNetworkDef.scala [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        .withName(name)
+        .withDriver(config.driver)
+        .withLabels(stableLabels.asJava)
+        .exec()
+      ContainerNetwork(name, network.getId)
+    }
+
+    override def acquire: F[ContainerNetwork[T]] = DIEffect[F].maybeSuspend {
+      if (config.reuse) {
+        val existedNetworks: List[Network] = client.listNetworksCmd().exec().asScala.toList
```
Comment (neko-kai) [inline distage/distage-framework-docker/src/main/scala/izumi/distage/docker/ContainerNetworkDef.scala]: I guess this needs file locking too. We should probably extract file locking helpers to a new class at this point

Comment (Caparow) [reply distage/distage-framework-docker/src/main/scala/izumi/distage/docker/ContainerNetworkDef.scala]: Yup, makes sense.

**Review by neko-kai — APPROVED** [review]
👍

**Pull request merged**